get_s3.go

```diff
@@ -173,7 +173,15 @@ func (g *S3Getter) getObject(ctx context.Context, client *s3.S3, dst, bucket, ke
 		return err
 	}
 
-	return copyReader(dst, resp.Body, 0666, g.client.umask())
+	body := resp.Body
+
+	if g.client.ProgressListener != nil {
```

Comment:
Watch out, the tests paniced here for me:
```suggestion
	if g.client != nil && g.client.ProgressListener != nil {
```

Reply:
Added.